### ISSUE TITLE
Fails to build on up-to-date Arch Linux due to CD-related function calls

### ISSUE THREAD

**Issue by turmoni** [adventuregamestudio/ags#459]
I'm hesitant to raise this bug since people seem to have successfully compiled since commit 4823c29604b9e5a776defb7eb4eff5b709f6813e , but I'm consistently failing to build the engine on my setup, with the following error:

```platform/linux/acpllnx.o
In file included from platform/linux/acpllnx.cpp:24:
../Engine/ac/runtime_defines.h:34: warning: "MAX_TIMERS" redefined
 #define MAX_TIMERS       21
 
In file included from /usr/include/xalleg.h:26,
                 from platform/linux/acpllnx.cpp:23:
/usr/include/allegro/internal/aintern.h:161: note: this is the location of the previous definition
 #define MAX_TIMERS      16
 
platform/linux/acpllnx.cpp: In member function ‘virtual int AGSLinux::CDPlayerCommand(int, int)’:
platform/linux/acpllnx.cpp:67:10: error: ‘cd_player_control’ was not declared in this scope
   return cd_player_control(cmdd, datt);
          ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make: *** [Makefile:46: platform/linux/acpllnx.o] Error 1
```

The relevant line:  https://github.com/adventuregamestudio/ags/blob/bd86e80343ec94934d9a501bd877744591e45ab4/Engine/platform/linux/acpllnx.cpp#L67

I'm wondering if this is perhaps compiler-specific (using GCC 8.1.1) and earlier releases didn't care about this. I did see that various other ports have these calls replaced with static return values, is this what's needed here?

**Comment by ivan-mogilko** [comment]
This is probably due big rewrite we are doing at master, as explained in #448.
Some of the deprecated functionality was removed from the code engine code, but the references could stay in port-related code. They simply need to be removed/disabled.

For the time being you could use "ags3" branch or "release-3.4.1" tag.

**Comment by AlanDrake** [comment]
Yes, this is my fault because I haven't tried building in other platforms beside windows.

**Comment by ivan-mogilko** [comment]
Sorry for the delay, I was busy working on another branch.
This problem should be fixed with 4a7779b30b1c2863ce998ab9bac343acbfc197b7
